### PR TITLE
Deduplicated code, eliminated manual array<->collection transfers.

### PR DESCRIPTION
**Pull request by tkroman**
// Unrelated to this pull-request:
@mateiz, I also took liberty to uncomment that line here: b21cbe688fdf5136b7d2569b1edc19fea117a8cc. Let me know if you consider it worth separate pull-request or should we wait?

UPD: Now it looks even more clearly duplicated for me, maybe I should try to generify it even more?

### CODE REVIEW

**Comment by andyk** [comment]
Jenkins, test this please.

**Comment by rxin** [comment]
@JoshRosen can you take a look at this?

**Comment by JoshRosen** [comment]
After a cursory look, this seems fine to me; I think the existing tests in https://github.com/mesos/spark/blob/f649dabb4a3a57cb25a852808297fb718cbfedd4/core/src/test/scala/spark/JavaAPISuite.java#L81 should cover this.

**Comment by tkroman** [comment]
So, maybe someone could order Jenkins to test this?

**Comment by mateiz** [comment]
Jenkins, retest this please

**Comment by mateiz** [comment]
Jenkins, retest this please

**Comment by mateiz** [comment]
Jenkins had some trouble yesterday but I'll test it manually if it doesn't work.

**Comment by mateiz** [comment]
Jenkins, retest this please

File: core/src/main/scala/spark/api/java/JavaSparkContext.scala [hunk 1]
```diff
@@ -258,27 +258,39 @@ class JavaSparkContext(val sc: SparkContext) extends JavaSparkContextVarargsWork
...
+  //   implicit val kcm: ClassManifest[K] = first.kManifest
+  //   implicit val vcm: ClassManifest[V] = first.vManifest
+  //   new JavaPairRDD(sc.union(rdds)(cm))(kcm, vcm)
+  // }
+
+  // /** Build the union of two or more RDDs. */
+  // override def union(first: JavaDoubleRDD, rest: java.util.List[JavaDoubleRDD]): JavaDoubleRDD = {
+  //   val rdds: Seq[RDD[Double]] = (Seq(first) ++ asScalaBuffer(rest)).map(_.srdd)
+  //   new JavaDoubleRDD(sc.union(rdds))
+  // }
```
Comment (mateiz) [inline core/src/main/scala/spark/api/java/JavaSparkContext.scala:293]: Actually, I do have one comment on the code: please don't leave commented-out lines in the project. If you don't need these versions of union(), delete them.

**Pull request closed without merging**